utils/src/main/java/software/amazon/awssdk/utils/ImmutableMap.java

```diff
@@ -198,8 +199,18 @@ public static <K, V> ImmutableMap<K, V> of(K k0, V v0, K k1, V v1,
 
     private static <K, V> void putAndWarnDuplicateKeys(Map<K, V> map, K key,
                                                        V value) {
+        putAndWarnDuplicateKeys(map, key, value, false);
+    }
+
+    private static <K, V> void putAndWarnDuplicateKeys(Map<K, V> map, K key, V value, boolean allowDuplicateKeys) {
```

Comment:
instead of overloading the  method can we remove the static and use
`this.allowDuplicateKeys` 
Also can we remove the warn logs completely since we are explicitly saying allow the duplicate keys it doesn't make sense logging warn when we explicitly say it , DEBUG logs makes sense .WDYT ?